### ISSUE TITLE
Height autodetect feature

### ISSUE THREAD

**Issue by ModernBalloonie** [Yellow-Dog-Man/Resonite-Issues#1652]
### Is your feature request related to a problem? Please describe.

The way of putting in your height (old and new settings menu) is a little bit weird sometimes because people sometimes don't know their actual height.

### Describe the solution you'd like

A way to auto-detect height by comparing where your headset is to the floor!

This would only work when standing up in your playspace.

### Describe alternatives you've considered

None really, besides getting a tape measure and actually measuring myself.

### Additional Context

I've seen this solution in VRchat, and it was really easy to use because you just hit one button, and bam! you're good!

### Requesters

modernballoonie on Discord!

**Comment by 5H4D0W-X** [comment]
This would also account for differences in haircut or head shapes that would lead to varying offsets between total height and the eye height that Resonite actually measures with the headset position

**Comment by amplified1** [comment]
This would also make swapping the headset between different people for demo purposes much easier. Auto-calibration is an extremely common feature across different VR games so I feel like it's something new users will expect and find odd if it isn't present.

**Comment by Frooxius** [comment]
It will need to be explicit, like a button press when you are known to be standing up to work most reliably.

Auto-calibration doesn't work the best, because you need to make lots of assumptions - is the user sitting? Are they slouching? Are they moving around? Are they standing up on their toes? You don't know what they're doing to determine which height is actually "correct" and making constant corrections will be jarring, as their avatar size and other stuff changes.

**Comment by ModernBalloonie** [comment]
> It will need to be explicit, like a button press when you are known to be standing up to work most reliably.
> 
> Auto-calibration doesn't work the best, because you need to make lots of assumptions - is the user sitting? Are they slouching? Are they moving around? Are they standing up on their toes? You don't know what they're doing to determine which height is actually "correct" and making constant corrections will be jarring, as their avatar size and other stuff changes.

Yeah I was thinking something along the lines of a button press myself.

**Comment by Frooxius** [comment]
We'd most likely just do the button option. The seated users can either just stand up or input their height, I don't know if it would be worth the complications.
